### PR TITLE
Update Drupal 10 Hub

### LINKED ISSUE #8008: Create a Drupal 10 Site Doc Update
Re: [Create a Drupal 10 Site](https://pantheon.io/docs/drupal-10)

Priority: Low/Medium/High (choose one, remove the other options)

## Suggested Resolution

### PR DESCRIPTION
**Pull request by wordsmither**
Closes #8008

## Summary

**[Drupal 10](https://pantheon.io/docs/drupal-10)** - added upgrade instructions, availibility information, and FAQ.

## Effect

The following changes are already committed:

*
*

## Remaining Work and Prerequisites

The following changes still need to be completed:

- [ ] List any outstanding work here

### Dependencies and Timing

- [ ] Other prerequisites that must be completed before merging this PR

**Release**:
- [ ] When ready
- [ ] After date: $DATE

--------------------------------------------------

## Post Launch

**Do not remove** - To be completed by the docs team upon merge:

- [ ] Redirect `/docs/old-path/` => `/docs/new-path/` (if applicable)
- [ ] Include/exclude pages ^ respectively within docs search service provider (if applicable)
- [ ] For Heroes - add a props post to the [discussion board](https://discuss.pantheon.io/c/pantheon-platform/documentation/17).
- [ ] Remove from the [project board](https://github.com/pantheon-systems/documentation/projects/14)

### CODE REVIEW

**Comment by wordsmither** [comment]
[Preview](https://pr-8009--pantheon-docs.my.pantheonfrontend.website/docs/drupal-10)
[Preview - landing](https://pr-8009--pantheon-docs.my.pantheonfrontend.website/docs/develop-drupal)

File: source/content/drupal-10.md
Comment (scottbuscemi) [inline source/content/drupal-10.md:26]: Let's remove this one for now.

**Review by stovak — APPROVED** [review]

File: source/content/drupal-10.md
Comment (stovak) [reply source/content/drupal-10.md:26]: Agree. We don't want to talk about "upgrading" yet.

File: source/content/drupal-10.md
Comment (greg-1-anderson) [inline source/content/drupal-10.md:33]: Have we tested with rc2 yet? Seems that we should be able to downgrade this warning / upgrade our confidence at this point.

File: source/content/drupal-10.md
Comment (greg-1-anderson) [inline source/content/drupal-10.md:39]: Drupal (Composer Managed) will be on PHP 8.1 later today. Let's change this to provide instructions for folks with existing sites, and tell them they should be good to go if they create a new site.

If the user sees an update offered on the dashboard to go to PHP 8.1, they should apply that, and it will do the right thing for them.

If they don't see that update, then maybe they are on a different upstream. They can follow steps 2 - 4 below (rewrite without step 1, so existing step 2 becomes new step 1).  They will also need to modify their [platform php configuration](https://github.com/pantheon-upstreams/drupal-composer-managed/blob/main/composer.json#L84) in their top-level composer.json to read "8.1.13" instead of "7.4". Add this as a new step at the end.

**Review by greg-1-anderson — CHANGES_REQUESTED** [review]
Let's rewrite the instructions for upgrading to PHP 8.1 to focus on new sites.

File: source/content/drupal-10.md
Comment (greg-1-anderson) [inline source/content/drupal-10.md:47]: The section that follows does not need the "Add the following line" step or the "Commit and push the change" step

Comment (greg-1-anderson) [reply source/content/drupal-10.md:47]: Note that the section I am referring to was not modified by this PR and is hidden by default on the GitHub diff

File: source/content/drupal-10.md
Comment (greg-1-anderson) [inline source/content/drupal-10.md:31]: Some sites won't have the update. It would be useful to include instructions for those sites. Or maybe just add another sentence linking them to the PHP Versions document.

**Review by greg-1-anderson — CHANGES_REQUESTED** [review]
Closer

**Comment by wordsmither** [comment]
DO NOT MERGE until 12/14

**Comment by whitneymeredith** [comment]
LGTM

**Pull request merged**
